pandas/tests/io/test_common.py

```diff
@@ -642,6 +643,21 @@ def close(self):
                 handles.created_handles.append(TestError())
 
 
+@td.skip_if_no("fsspec")
+def test_read_csv_chained_url_no_error():
+    # GH 60100
+    tar_file = "pandas/tests/io/data/tar/test-csv.tar"
+
+    x = pd.read_csv(f"tar://test.csv::file://{tar_file}", compression=None)
+    y = pd.read_csv(f"tar://test.csv::file://{tar_file}", compression="infer")
+
+    x_to_json_expected_output = '{"1;2":{"0":"3;4"}}'
+    y_to_json_expected_output = '{"1;2":{"0":"3;4"}}'
```

Comment:
Can you just create the expected DataFrame and use `tm.assert_frame_equal`. Conversion JSON is lossy.